### PR TITLE
Fix 2 lock-order-inversion bugs as found by TSan on m-c tests

### PR DESCRIPTION
**Pull request by badboy**
(Taking this for a try run in m-c to ensure it works before we land this)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # Unreleased changes
 
-* Rust
-  * Introduce the Memory Distribution metric type in the RLB. [#1376](https://github.com/mozilla/glean/pull/1376)
-
 [Full changelog](https://github.com/mozilla/glean/compare/v33.7.0...main)
 
 * Rust
-  * Shut down Glean in tests before resetting to make sure they don't mistakenly init Glean twice in parallel.
+  * Introduce the Memory Distribution metric type in the RLB. [#1376](https://github.com/mozilla/glean/pull/1376)
```
Comment (badboy) [inline CHANGELOG.md:6]: I went ahead and fixed the changelog while I was here.

**Comment by badboy** [comment]
m-c try run: https://treeherder.mozilla.org/jobs?repo=try&revision=e5a2ac50a73b1d2dd2f3475847be268961d6dca5

**Review by Dexterp37 — APPROVED** [review]
Great catches. Would you consider filing a bug for discussing enabling TSan test coverage for the glean SDK?

**Comment by badboy** [comment]
> Great catches. Would you consider filing a bug for discussing enabling TSan test coverage for the glean SDK?

lock-order-inversion

**Comment by badboy** [comment]
it's green!

**Pull request merged**